zcash_client_backend/src/data_api/error.rs

```diff
@@ -54,6 +54,9 @@ pub enum Error<DataSourceError, CommitmentTreeError, SelectionError, FeeError, N
     /// It is forbidden to provide a memo when constructing a transparent output.
     MemoForbidden,
 
+    /// Attempted to create a spend to an unsupported output type (i.e. Orchard)
+    UnsupportedOutputType,
```

Comment:
This should wrap a `PoolType` so that the  user can observe the failure.

Reply:
Also, this addition will need to be documented in `zcash_client_backend/CHANGELOG.md`